src/main/java/uk/nhs/hee/tis/trainee/sync/service/TcsSyncService.java

```diff
@@ -83,17 +88,19 @@
     this.restTemplate = restTemplate;
     this.personService = personService;
 
-    tableNameToMappingFunction = Map.of(
-        TABLE_CONTACT_DETAILS, mapper::toContactDetails,
-        TABLE_GDC_DETAILS, mapper::toGdcDetailsDto,
-        TABLE_GMC_DETAILS, mapper::toGmcDetailsDto,
-        TABLE_PERSON, mapper::toBasicDetailsDto,
-        TABLE_PERSON_OWNER, mapper::toPersonOwnerDto,
-        TABLE_PERSONAL_DETAILS, mapper::toPersonalInfoDto,
-        TABLE_QUALIFICATION, mapper::toQualificationDto,
-        TABLE_PLACEMENT, mapper::toPlacementDto,
-        TABLE_PROGRAMME_MEMBERSHIP, mapper::toProgrammeMembershipDto,
-        TABLE_CURRICULUM, mapper::toCurriculumDto
+    tableNameToMappingFunction = Map.ofEntries(
+        Map.entry(TABLE_CONTACT_DETAILS, mapper::toContactDetails),
+        Map.entry(TABLE_GDC_DETAILS, mapper::toGdcDetailsDto),
+        Map.entry(TABLE_GMC_DETAILS, mapper::toGmcDetailsDto),
+        Map.entry(TABLE_PERSON, mapper::toBasicDetailsDto),
+        Map.entry(TABLE_PERSON_OWNER, mapper::toPersonOwnerDto),
+        Map.entry(TABLE_PERSONAL_DETAILS, mapper::toPersonalInfoDto),
+        Map.entry(TABLE_QUALIFICATION, mapper::toQualificationDto),
+        Map.entry(TABLE_PLACEMENT, mapper::toPlacementDto),
+        Map.entry(TABLE_PROGRAMME_MEMBERSHIP, mapper::toProgrammeMembershipDto),
+        Map.entry(TABLE_CURRICULUM, mapper::toCurriculumDto),
+        Map.entry(TABLE_PLACEMENT_SPECIALTY, mapper::toPlacementSpecialtyDto),
+        Map.entry(TABLE_SPECIALTY, mapper::toSpecialtyDto)
```

Comment:
Are these ever used?  I can't see any calls to the this class from the enricher for a `PlacementSpecialty` or `Specialty`.

Reply:
I can't see them being used so I deleted these. Also, in **TraineeDetailsMapper** I deleted the methods for a `Specialty` and a `PlacementSpecialty`, since these aren't models in **tis-trainee-details**, but just one field of `Placement` called "specialty".

Reply:
Yes, sorry, me again as per the previous conversation